core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/HttpWorkspaceManagerDAO.scala

```diff
@@ -57,39 +57,21 @@ class HttpWorkspaceManagerDAO(apiClientProvider: WorkspaceManagerApiClientProvid
                                                displayName: String,
                                                spendProfileId: String,
                                                billingProjectNamespace: String,
+                                               applicationIds: Seq[String],
+                                               policyInputs: Option[WsmPolicyInputs],
                                                ctx: RawlsRequestContext
-  ): CreatedWorkspace =
-    getWorkspaceApi(ctx).createWorkspace(
-      new CreateWorkspaceRequestBody()
-        .id(workspaceId)
-        .displayName(displayName)
-        .spendProfile(spendProfileId)
-        .stage(WorkspaceStageModel.MC_WORKSPACE)
-        .projectOwnerGroupId(billingProjectNamespace)
-    )
-
-  override def createProtectedWorkspaceWithSpendProfile(workspaceId: UUID,
-                                                        displayName: String,
-                                                        spendProfileId: String,
-                                                        billingProjectNamespace: String,
-                                                        ctx: RawlsRequestContext
   ): CreatedWorkspace = {
-    val policyInputs = new WsmPolicyInputs()
-    val protectedPolicyInput = new WsmPolicyInput()
-    protectedPolicyInput.name("protected-data")
-    protectedPolicyInput.namespace("terra")
-    protectedPolicyInput.additionalData(List().asJava)
-
-    policyInputs.addInputsItem(protectedPolicyInput)
-    getWorkspaceApi(ctx).createWorkspace(
-      new CreateWorkspaceRequestBody()
-        .id(workspaceId)
-        .displayName(displayName)
-        .spendProfile(spendProfileId)
-        .stage(WorkspaceStageModel.MC_WORKSPACE)
-        .policies(policyInputs)
-        .projectOwnerGroupId(billingProjectNamespace)
-    )
+    val request = new CreateWorkspaceRequestBody()
+      .id(workspaceId)
+      .displayName(displayName)
+      .spendProfile(spendProfileId)
+      .stage(WorkspaceStageModel.MC_WORKSPACE)
+      .projectOwnerGroupId(billingProjectNamespace)
+      .applicationIds(applicationIds.asJava)
+
+    policyInputs.map(request.policies)
```

Comment:
This looks confusing to me-- it populates the request with the policies? Is it somehow handling the Option?

Reply:
This is equivalent to:
```
policyInputsOpt.map { actualPolicyInputs =>
    request.policies(actualPolicyInputs)
}
```
(I renamed `policyInputs` here for clarity)


Calling `.map` on an `Option` will only execute the code inside the block if the `Option` is defined. In this case, we'll only set the request policies if `policyInputs` is defined. 

You can also omit the `()` and parameter if the function called inside the `.map` accepts the same type that is inside the `Option` which is why `.map(request.policies)` is okay.